src/zjs_ble.c

```diff
@@ -375,14 +422,17 @@ static ZJS_DECL_FUNC(zjs_ble_update_value_callback_function)
     // args: buffer
     ZJS_VALIDATE_ARGS(Z_BUFFER);
 
-    if (ble_conn.default_conn) {
-        uintptr_t ptr;
-        if (jerry_get_object_native_handle(this, &ptr)) {
-            ble_characteristic_t *chrc = (ble_characteristic_t *)ptr;
-            if (chrc->chrc_attr) {
-                zjs_buffer_t *buf = zjs_buffer_find(argv[0]);
-                bt_gatt_notify(ble_conn.default_conn, chrc->chrc_attr,
+    uintptr_t ptr;
+    if (jerry_get_object_native_handle(this, &ptr)) {
+        ble_characteristic_t *chrc = (ble_characteristic_t *)ptr;
+        if (chrc->chrc_attr) {
+            zjs_buffer_t *buf = zjs_buffer_find(argv[0]);
```

Comment:
Looking into it more myself, no it looks like this is directly called by JS so you can't be sure what they'll pass in. So you need to check whether buf is NULL here.